server/plugin.go

```diff
@@ -494,16 +500,26 @@ func (p *Plugin) OnActivate() error {
 			func() []byte { return []byte(p.configuration.EncryptionKey) },
 		)
 		p.store = timerlayer.New(store, p.GetMetrics())
+
+		defer func() {
+			if err != nil {
+				if err = p.store.Shutdown(); err != nil {
+					p.API.LogWarn("failed to close db connection", "error", err)
```

Comment:
Should this be an error like when it fails above? We /might/ want to take action in this case and rotate the pods.